### PR TITLE
25 toggle private mode

### LINKED ISSUE #25: Toggle private mode
Estimate: 3 

Feature: Toggle private mode
    As a user
    I want to click a button in the RichClipboard window
    In order to toggle logging clipboard activity

Scenario: User enters private mode and copies some text
        Given that I have RichClipboard open
        And private mode is toggled “on”
        When I copy text to the clipboard
        Then no new clipping is displayed in RichClipboard

Scenario: User leaves private mode and copies some text
        Given that I have RichClipboard open
        And private mode is toggled “off”
        When I copy text to the clipboard
        Then a new clipping is displayed in RichClipboard

### LINKED ISSUE #27: TEST: Toggle private mode

### PR DESCRIPTION
**Pull request by martingraf4**
Closes #25 
Closes #27 

- [x] High Test-Coverage (90%), GUI Code might be an exception
- [x] All Tests passing
- [ ] All files autoformatted (to be discussed)
- [x] Codequality: not many long methods
- [x] Codequality: Encapsulation (classes should be loosely coupled, law of demeter,...)
- [x] Timetracking: Issue Timetracking is complete (Time spent)

### CODE REVIEW

File: RichClipboard.package/RCClippingsWindow.class/instance/createPrivateModeButton.st [hunk 1]
```diff
@@ -0,0 +1,10 @@
+private mode
+createPrivateModeButton
+	| button |
```
Comment (zortax) [inline RichClipboard.package/RCClippingsWindow.class/instance/createPrivateModeButton.st:3]: new line missing
```suggestion
	| button |
	
```

File: RichClipboard.package/RCClippingsWindow.class/instance/createPrivateModeButton.st [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+	button := PluggableButtonMorphPlus
+				on: self
+				getState: #privateModeButtonEnabled
+				action: #togglePrivateMode
+				label: #privateModeButtonLabel.
+	button height: self class defaultBottomBarHeight - 4.
```
Comment (zortax) [inline RichClipboard.package/RCClippingsWindow.class/instance/createPrivateModeButton.st:10]: new line missing
```suggestion
	button height: self class defaultBottomBarHeight - 4.
	
```

File: RichClipboard.package/RCClippingsWindow.class/instance/togglePrivateMode.st [hunk 3]
```diff
@@ -0,0 +1,5 @@
+private mode
+togglePrivateMode
+	self privateModeActive
```
Comment (zortax) [inline RichClipboard.package/RCClippingsWindow.class/instance/togglePrivateMode.st]: Coding standards!

File: RichClipboard.package/RCClippingsWindow.class/instance/turnOffPrivateMode.st [hunk 4]
```diff
@@ -0,0 +1,8 @@
+private mode
+turnOffPrivateMode
+	self privateModeActive
+		ifTrue: [
```
Comment (zortax) [inline RichClipboard.package/RCClippingsWindow.class/instance/turnOffPrivateMode.st]: würde das ifTrue nicht in die neue Zeile packen

File: RichClipboard.package/RCClippingsWindow.class/instance/turnOnPrivateMode.st [hunk 5]
```diff
@@ -0,0 +1,8 @@
+private mode
+turnOnPrivateMode
+	self privateModeActive
+		ifFalse: [
```
Comment (zortax) [inline RichClipboard.package/RCClippingsWindow.class/instance/turnOnPrivateMode.st]: same here

**Review by zortax — CHANGES_REQUESTED** [review]

File: RichClipboard.package/RCClippingsWindow.class/instance/turnOffPrivateMode.st [hunk 6]
```diff
@@ -0,0 +1,8 @@
+private mode
+turnOffPrivateMode
+	self privateModeActive
+		ifTrue: [ 
+			self storage privateModeActive: false.
+			privateModeActive := false.
+			self updateWindowTitle
+		]
```
Comment (zortax) [inline RichClipboard.package/RCClippingsWindow.class/instance/turnOffPrivateMode.st]:
```suggestion
	self privateModeActive ifTrue: [ 
		self storage privateModeActive: false.
		privateModeActive := false.
		self updateWindowTitle
	]
```

File: RichClipboard.package/RCClippingsWindow.class/instance/turnOnPrivateMode.st [hunk 7]
```diff
@@ -0,0 +1,8 @@
+private mode
+turnOnPrivateMode
+	self privateModeActive
+		ifFalse: [ 
+			self storage privateModeActive: true.
+			privateModeActive := true.
+			self updateWindowTitle
+		]
```
Comment (zortax) [inline RichClipboard.package/RCClippingsWindow.class/instance/turnOnPrivateMode.st]:
```suggestion
	self privateModeActive ifFalse: [ 
		self storage privateModeActive: true.
		privateModeActive := true.
		self updateWindowTitle
	]
```

File: RichClipboard.package/RCClippingsWindow.class/instance/togglePrivateMode.st [hunk 8]
```diff
@@ -0,0 +1,5 @@
...
+		ifTrue: [ self turnOffPrivateMode ]
+		ifFalse: [ self turnOnPrivateMode ]
```
Comment (zortax) [inline RichClipboard.package/RCClippingsWindow.class/instance/togglePrivateMode.st]:
```suggestion
	self privateModeActive ifTrue: [
		self turnOffPrivateMode
	] ifFalse: [
		self turnOnPrivateMode
	]
```

**Review by zortax — CHANGES_REQUESTED** [review]

**Review by WolfgangDaniel — DISMISSED** [review]

**Review by zortax — APPROVED** [review]

**Review by WolfgangDaniel — APPROVED** [review]

**Pull request merged**